flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultResourceAllocationStrategyTest.java

```diff
@@ -138,6 +137,51 @@ void testFulfillRequirementWithRegisteredResourcesEvenly() {
                                         .isTrue());
     }
 
+    @Test
+    void testFullFillRequirementWithMinimumPossibleRegisteredResources() {
+        final TaskManagerInfo taskManager1 =
+                new TestingTaskManagerInfo(
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE);
+        final TaskManagerInfo taskManager2 =
+                new TestingTaskManagerInfo(
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE);
+        final JobID jobId = new JobID();
+        final List<ResourceRequirement> requirements = new ArrayList<>();
+        final ResourceProfile largeResource = DEFAULT_SLOT_RESOURCE.multiply(2);
+        final TaskManagerResourceInfoProvider taskManagerResourceInfoProvider =
+                TestingTaskManagerResourceInfoProvider.newBuilder()
+                        .setRegisteredTaskManagersSupplier(
+                                () -> Arrays.asList(taskManager1, taskManager2))
+                        .build();
+        requirements.add(ResourceRequirement.create(largeResource, 4));
+        requirements.add(ResourceRequirement.create(ResourceProfile.UNKNOWN, 2));
+
+        final ResourceAllocationResult result =
+                MIN_RESOURCES_STRATEGY.tryFulfillRequirements(
+                        Collections.singletonMap(jobId, requirements),
+                        taskManagerResourceInfoProvider,
+                        resourceID -> false);
+
+        assertAllSlotsAllocatedToRegisteredTaskManagersOnly(result);
+
+        // Assert only 1 TM is used to fulfill the requirements of 6 slots
```

Comment:
nit: If possible, using assert messages instead of comments makes the comment more accessible (i.e. also available in the assertion call).